mypyc/lib-rt/CPy.h

```diff
@@ -456,6 +456,14 @@ static inline bool CPyFloat_Check(PyObject *o) {
     return PyFloat_Check(o) || PyLong_Check(o);
 }
 
+static inline bool CPy_TypeCheck(PyObject *o, PyObject *type) {
```

Comment:
Add TODO comment about figuring out how to avoid the inline function in non-C back ends that may not be able to use the inline function. This may be worth creating an issue about, since this is performance-critical, I think.